finality-aleph/src/network/mod.rs

```diff
@@ -43,6 +43,21 @@ pub mod testing {
     };
 }
 
+/// This trait is needed for logging. It implements a shorter version of PeerId for ids implementing display.
```

Comment:
This has... very weird assumptions that are not at all reflected in the type.

I would suggest something like adding this method to the `PeerId` trait explicitly as `short_id` and in the implementations perhaps use anothe function from `crypto` which would only be defined for specific length slices (which we could get from both kinds of `PeerId` in our case).

And then we could have a `struct Shortened<P: PeerId>(P)` wrapper with it's own implementation of `Display`, which would use this function. You would still have to wrap this at some point, but you could do that in the constructor of the status report, making printing simpler.

Reply:
I'm not so sure about wrapping here just for logging sake. This trait only needs `PeerId` to implement display and nothing else. If you worry about a constraint on it having some length we can add a check here if PeerId is shorter then 13 characters and print it whole in this case.

 We can also just move `short_id` inside `PeerId` trait. This another trait is completely unnecessary.

Reply:
Adding `short_id` to `PeerId` with a default implementation as now, plus a check for 13 chars sounds good to me. The name could be better, to make clear it's for printing/logging, but I don't have any great ideas for that – `pretty` is too generic, `pretty_short` just sounds weird, `short_n_sweet` is too much of a joke even for me, `short_display` uncovers how terribly we are mangling our usage of `Display`...

Another related idea, although I haven't checked it works with `AuthorityId` – maybe instead of `Display` demand that `PeerId` implements `ToString` and then use that in the default implementation described above. This is kinda what we are already doing for substrate peer ids, and it feels cleaner than overriding `Display` behaviour. This is a very soft suggestion though, just doing what the first sentence of this comment says should be enough.

Reply:
I think in rust `ToString` Is be default implemented using `Display` and documentation says Display should be implemented instead because of that.

I would leave Display here and maybe rename it to `to_short_string`? Technically in is `to_string` but another `to_string` which is shorter...